From ecd84bd29fd7eff2b8db4c04e92224929322a51f Mon Sep 17 00:00:00 2001
From: wencyu <yuwenchao@qiyi.com>
Date: Tue, 29 Sep 2020 14:42:52 +0800
Subject: [PATCH 1/2] add debug log for ixgbe fdir setting

---
 drivers/net/ixgbe/ixgbe_fdir.c | 23 +++++++++++++++++++++++
 1 file changed, 23 insertions(+)

diff --git a/drivers/net/ixgbe/ixgbe_fdir.c b/drivers/net/ixgbe/ixgbe_fdir.c
index e559f0f..c17642a 100644
--- a/drivers/net/ixgbe/ixgbe_fdir.c
+++ b/drivers/net/ixgbe/ixgbe_fdir.c
@@ -278,6 +278,14 @@ static void ixgbe_fdir_stats_get(struct rte_eth_dev *dev,
 	uint32_t fdiripv6m; /* IPv6 source and destination masks. */
 	volatile uint32_t *reg;
 
+	PMD_INIT_LOG(DEBUG, "%s: tci 0x%04x, ip4src 0x%08x, ip4dst 0x%08x, "
+			"ip6src 0x%04x, ip6dst 0x%04x, src_port 0x%04x, dst_port 0x%04x, "
+			"flex 0x%04x, mac_addr 0x%02x, tunid 0x%08x, tuntype 0x02%x\n", __func__,
+			info->mask.vlan_tci_mask, info->mask.src_ipv4_mask, info->mask.dst_ipv4_mask,
+			info->mask.src_ipv6_mask, info->mask.dst_ipv6_mask, info->mask.src_port_mask,
+			info->mask.dst_port_mask, info->mask.flex_bytes_mask, info->mask.mac_addr_byte_mask,
+			info->mask.tunnel_id_mask, info->mask.tunnel_type_mask);
+
 	PMD_INIT_FUNC_TRACE();
 
 	/*
@@ -1242,6 +1250,21 @@ static void ixgbe_fdir_stats_get(struct rte_eth_dev *dev,
 	struct ixgbe_fdir_filter *node;
 	bool add_node = FALSE;
 
+	PMD_DRV_LOG(DEBUG, "%s: ixgbe_fdir_rule: b_spec %d, b_mask %d, mode %d, flags 0x%08x, softid %d, "
+			"queue %d, flex_off %d....ixgbe_fdir: vm_pool %d, flow_type %d, vlan_id %d, dst_ip 0x%8x, src_ip "
+			"0x%8x, inner_mac %02x:%02x:%02x:%02x:%02x:%02x, tuntype 0x%4x, tni_vni 0x%08x, src_port 0x%04x, "
+			"dst_port 0x%04x, flexbytes %d, bkt_hash %d\n", __func__, rule->b_spec, rule->b_mask,
+			rule->mode, rule->fdirflags, rule->soft_id, rule->queue, rule->flex_bytes_offset,
+			rule->ixgbe_fdir.formatted.vm_pool, rule->ixgbe_fdir.formatted.flow_type,
+			rule->ixgbe_fdir.formatted.vlan_id,
+			*((uint32_t *)&rule->ixgbe_fdir.formatted.dst_ip[0]),
+			*((uint32_t *)&rule->ixgbe_fdir.formatted.src_ip[0]),
+			rule->ixgbe_fdir.formatted.inner_mac[0], rule->ixgbe_fdir.formatted.inner_mac[1],
+			rule->ixgbe_fdir.formatted.inner_mac[2], rule->ixgbe_fdir.formatted.inner_mac[3],
+			rule->ixgbe_fdir.formatted.inner_mac[4], rule->ixgbe_fdir.formatted.inner_mac[5],
+			rule->ixgbe_fdir.formatted.tunnel_type, rule->ixgbe_fdir.formatted.tni_vni,
+			rule->ixgbe_fdir.formatted.src_port, rule->ixgbe_fdir.formatted.dst_port,
+			rule->ixgbe_fdir.formatted.flex_bytes, rule->ixgbe_fdir.formatted.bkt_hash);
 	if (fdir_mode == RTE_FDIR_MODE_NONE ||
 	    fdir_mode != rule->mode)
 		return -ENOTSUP;
-- 
1.8.3.1

